### PR TITLE
Add IFrame and remove requests_html

### LINKED ISSUE #842: use IFrame to display embedded D3 plot
when generating a pipeline plot with the D3 backend with `output='embed'`:

https://github.com/ploomber/ploomber/blob/77d7dbf6b143ee17f886f76b6f3163d0ad5a39c9/src/ploomber/dag/dag.py#L893

we use the requests-html library to load the HTML file since we need to execute the embedded JS for the plot to display:

https://github.com/ploomber/ploomber/blob/77d7dbf6b143ee17f886f76b6f3163d0ad5a39c9/src/ploomber/dag/plot.py#L66

and then we render the plot with a `IPython.display.HTML` object, this allows us to embed the plot in Jupyter. However, as pointed in https://github.com/ploomber/projects/issues/42 IPython has an `IFrame` object that correctly displays HTML and runs JS.

We should remove the requests-html code and substitute it for using `IFrame`

### PR DESCRIPTION
**Pull request by imskr**
## Describe your changes

- Add `IFrame` and remove `requests_html` to display embedded D3 plot.
## Issue ticket number and link
Closes #842 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests (when necessary).
- [ ] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/ploomber/dag/plot.py [hunk 1]
```diff
@@ -72,22 +70,8 @@ def with_d3(graph, output):
           pip_names=['requests-html', 'nest_asyncio'])
```
Comment (edublancas) [inline src/ploomber/dag/plot.py:70]: we dont need this decorator anymore

File: src/ploomber/dag/plot.py [hunk 2]
```diff
@@ -72,22 +70,8 @@ def with_d3(graph, output):
...
 def embedded_html(path):
     import asyncio
```
Comment (edublancas) [inline src/ploomber/dag/plot.py:72]: we dont need this

File: src/ploomber/dag/plot.py [hunk 3]
```diff
@@ -72,22 +70,8 @@ def with_d3(graph, output):
...
+    from IPython.display import IFrame
```
Comment (edublancas) [inline src/ploomber/dag/plot.py:73]: move import to the top

File: src/ploomber/dag/plot.py [hunk 4]
```diff
@@ -72,22 +70,8 @@ def with_d3(graph, output):
...
-    from requests_html import HTML as HTML_
-    from requests_html import AsyncHTMLSession
-
-    session = AsyncHTMLSession()
-    html = HTML_(html=Path(path).read_text(), session=session)
-    await html.arender()
-    # ensure we close the session, otherwise this will fail on windows
-    await session.close()
-    return HTML(data=html.html)
+    return asyncio.get_event_loop().run_until_complete(IFrame(src=path))
```
Comment (edublancas) [inline src/ploomber/dag/plot.py:77]: we dont need any of these, just return IFrame

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by imskr** [comment]
Thanks @edublancas I've made the changes as suggested :)

**Comment by edublancas** [comment]
can you fix the tests? looks like you're missing some arguments

```

    def embedded_html(path):
>       return IFrame(src=path)
E       TypeError: __init__() missing 2 required positional arguments: 'width' and 'height'

```

**Comment by edublancas** [comment]
closing due to inactivity

**Pull request closed without merging**